ros2_to_serial_bridge:
    ros__parameters:
        device: /dev/pts/6             # Which device to use for communications.
        baudrate: 0                    # The baudrate to set on the device; if
                                       # set to 0, setting baudrate is skipped.
        dynamic_serial_mapping_ms: 500 # How much time to wait to get dynamic
                                       # ROS2-to-serial mapping from the serial
                                       # side.  If less than 0, dynamic mapping
                                       # is disabled and only the topics below
                                       # will be setup.  If exactly 0, wait
                                       # forever for the other side to respond.
                                       # If greater than 0, wait that many
                                       # milliseconds for a response before
                                       # failing.  If this config is >= 0, the
                                       # any topics configured below will be
                                       # ignored.
        serial_protocol: px4           # The serial protocol to use.  Currently
                                       # available protocols are 'px4'
                                       # and 'cobs'.
        topics:                        # The list of topics to map; only used
                                       # if dynamic_serial_mapping_ms is < 0.
            chatter:
                serial_mapping: 9
                type: std_msgs/String
                direction: SerialToROS2
            uint16topic:
                serial_mapping: 12
                type: std_msgs/UInt16
                direction: SerialToROS2
            another:
                serial_mapping: 13
                type: std_msgs/String
                direction: ROS2ToSerial
